carma_wm/include/carma_wm/WMTestLibForGuidance.h

```diff
@@ -0,0 +1,458 @@
+#include <carma_wm/Geometry.h>
+/**
+ * This is a test library made for guidance unit tests. It includes following in general:
+ * - Helper functions to create the world from scratch or extend the world in getGuidanceTestMap()
+ * - addObstacle at a specified Cartesian or Trackpos point relative to specified lanelet Id
+ * - set route by giving series of lanelet Id in the map (setRouteById)
+ * - set speed of entire road (setSpeedLimit)
+ * - getGuidanceTestMap gives a simple one way, 3 lane map (25mph speed limit) with one static prebaked obstacle and 
+ *      4 lanelets in a lane (if 2 stripes make up one lanelet):
+ *
+ *        |1203|1213|1223|
+ *        | _  _  _  _  _|
+ *        |1202| Ob |1222|
+ *        | _  _  _  _  _|
+ *        |1201|1211|1221|    num   = lanelet id hardcoded for easier testing
+ *        | _  _  _  _  _|    |     = lane lines
+ *        |1200|1210|1220|    - - - = Lanelet boundary
+ *        |              |    O     = Default Obstacle
+ *        ****************
+ *           START_LINE
+ * - NOTE: please look at unit test on example how to use them
+ */
+namespace carma_wm
+{
+namespace test
+{
+using namespace lanelet::units::literals;
+
+enum TestMapOption {DEFAULT_OBSTACLE, NO_OBSTACLE, DEFAULT_SPEED_LIMIT, NO_SPEED_LIMIT};
+inline lanelet::Point3d getPoint(double x, double y, double z)
+{
+  return lanelet::Point3d(lanelet::utils::getId(), x, y, z);
+}
+
+inline lanelet::BasicPoint2d getBasicPoint(double x, double y)
+{
+  return lanelet::utils::to2D(getPoint(x, y, 0.0)).basicPoint();
+}
+
+// Defaults to double solid line on left and a solid line on right
+inline lanelet::Lanelet getLanelet(lanelet::LineString3d& left_ls, lanelet::LineString3d& right_ls,
+                                   const lanelet::Attribute& left_sub_type = lanelet::AttributeValueString::SolidSolid,
+                                   const lanelet::Attribute& right_sub_type = lanelet::AttributeValueString::Solid)
+{
+  left_ls.attributes()[lanelet::AttributeName::Type] = lanelet::AttributeValueString::LineThin;
+  left_ls.attributes()[lanelet::AttributeName::Subtype] = left_sub_type;
+
+  right_ls.attributes()[lanelet::AttributeName::Type] = lanelet::AttributeValueString::LineThin;
+  right_ls.attributes()[lanelet::AttributeName::Subtype] = right_sub_type;
+
+  lanelet::Lanelet ll;
+  ll.setId(lanelet::utils::getId());
+  ll.setLeftBound(left_ls);
+  ll.setRightBound(right_ls);
+
+  ll.attributes()[lanelet::AttributeName::Type] = lanelet::AttributeValueString::Lanelet;
+  ll.attributes()[lanelet::AttributeName::Subtype] = lanelet::AttributeValueString::Road;
+  ll.attributes()[lanelet::AttributeName::Location] = lanelet::AttributeValueString::Urban;
+  ll.attributes()[lanelet::AttributeName::OneWay] = "yes";
+  ll.attributes()[lanelet::AttributeName::Dynamic] = "no";
+
+  return ll;
+}
+
+inline lanelet::Lanelet getLanelet(std::vector<lanelet::Point3d> left, std::vector<lanelet::Point3d> right,
```

Comment:
Just like the function below, this function should have another version that takes the id